### PR TITLE
create and implement decorators

### PR DESCRIPTION
**Pull request by sireclifford**
- Create & implement decorators

> Base decorator

> Trimmer decorator

> Capitalize decorator

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,9 +1,10 @@
+require_relative 'nameable'
 # Person class
 class Person
```
Comment (Ugboaja-Uchechi) [inline person.rb]: - [ ] Great job 👍 . Kindly make sure your `Person` class inherits from `Nameable` class as stated in the project requirements 🙏

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
```
Comment (Ugboaja-Uchechi) [inline decorator.rb:5]: - [ ] In the constructor kindly assign a nameable object from params to an instance variable as stated in the project requirement 🙏

**Review by Ugboaja-Uchechi — CHANGES_REQUESTED** [review]
Hi @sireclifford,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- Linters are passing ✔️ 
- README is descriptive 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sireclifford** [comment]
Hello reviewer. 
Considering the review made by the previous reviewer, there must be a misunderstanding. I have checked the project requirements and the work of colleagues. The comment requesting for instance variable assignment isn't right. What I did before is the right one.

Should you notice a misunderstanding on my part, please explain/ suggest a better way to modify my work.
Thank you.

**Review by BenMukebo — APPROVED** [review]
### Hi @sireclifford, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- Profectional README file. ✔️  


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**